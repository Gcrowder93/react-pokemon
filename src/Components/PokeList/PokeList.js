import React from 'react';
import { Button } from '@mui/material';
export default function PokeList({ pokemon, currentPage, setCurrentPage, setLoading }) {
  console.log(pokemon);
  const handleNextPage = () => {
    setCurrentPage((prevState) => ++prevState);
    setLoading(true);
  };
  {
    return (
      <div className="container">
        <div className="grid">
          {pokemon.map((p) => (
            <div className="pokename" key={p._id}>
              console.log(p._id)
              {p.pokemon.charAt(0).toUpperCase() + p.pokemon.slice(1)}
              <img src={p.url_image}></img>
              <div>
                Type1:{p.type_1.charAt(0).toUpperCase() + p.type_1.slice(1)} Type2:
                {p.type_2.charAt(0).toUpperCase() + p.type_2.slice(1)}
                HP:{p.hp} ATT:{p.attack} DEF:{p.defense}
                <br></br> {/* Spec. Att: {p.special_attack} Spec. Def: {p.special_defence} */}
                {/* <div>
                    <br></br>
                  </div>{' '}
                  Speed {p.speed} */}
                <p>#{p.species_id}</p>
              </div>
            </div>
          ))}
        </div>
        <div className="pagenumber">Page: {currentPage}</div>
        <Button
          style={{
            backgroundColor: 'pink',
          }}
          variant="contained"
          className="button2"
          onClick={handleNextPage}
        >
          ◓Next Page◓
        </Button>
      </div>
    );
  }
}
