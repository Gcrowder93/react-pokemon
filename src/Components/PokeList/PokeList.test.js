import { render } from '@testing-library/react';
import PokeList from './PokeList';

test('renders the pokemon name', () => {
  const Charmander = {
    pokemon: 'Charmander',
    type_1: 'Fire',
    type_2: 'NA',
    _id: 1,
  };
  const container = render(<PokeList pokemon={[Charmander]} />);
  expect(container).toMatchSnapshot();
});
