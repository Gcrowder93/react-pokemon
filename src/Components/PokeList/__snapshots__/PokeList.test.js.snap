// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the pokemon name 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="grid"
        >
          <div
            class="pokename"
          >
            console.log(p._id)
            Charmander
            <img />
            <div>
              Type1:
              Fire
               Type2:
              NA
              HP:
               ATT:
               DEF:
              <br />
               
              <p>
                #
              </p>
            </div>
          </div>
        </div>
        <div
          class="pagenumber"
        >
          Page: 
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button2 css-sghohy-MuiButtonBase-root-MuiButton-root"
          style="background-color: pink;"
          tabindex="0"
          type="button"
        >
          ◓Next Page◓
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="grid"
      >
        <div
          class="pokename"
        >
          console.log(p._id)
          Charmander
          <img />
          <div>
            Type1:
            Fire
             Type2:
            NA
            HP:
             ATT:
             DEF:
            <br />
             
            <p>
              #
            </p>
          </div>
        </div>
      </div>
      <div
        class="pagenumber"
      >
        Page: 
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root button2 css-sghohy-MuiButtonBase-root-MuiButton-root"
        style="background-color: pink;"
        tabindex="0"
        type="button"
      >
        ◓Next Page◓
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
